ducktape/tests/test.py

```diff
@@ -150,16 +152,15 @@ def __init__(self, session_context, module=None, cls=None, function=None, inject
         # dict for toggling service log collection on/off
         self.log_collect = {}
 
-        # Individual test results go here
-        mkdir_p(self.results_dir)
-
-        self._logger_configured = False
-        self.configure_logger()
-
     def __repr__(self):
         return "<module=%s, cls=%s, function=%s, injected_args=%s>" % \
                (self.module, self.cls_name, self.function_name, str(self.injected_args))
 
+    def copy(self, **kwargs):
+        ctx_copy = TestContext(self.session_context, self.module, self.cls, self.function, self.injected_args, self.ignore)
```

Comment:
You should be able to avoid always having to keep this up to date with the constructor args by just using the fact that you can get the `__dict__`. You could either change all arguments in the constructor to be keyword arguments and then do something like:

`ctx_copy = TestContext(**self.__dict__)`

In fact, I think that works even if you leave `session_context` as is. Regardless, my previous comment about not including `ignore` on every `TestContext` objects would help here by keeping the number of arguments on this line fixed. As it stands, as we add more annotations this list is just going to get ridiculously unwieldy.